### ISSUE TITLE
log.Fatal() not printing anything

### ISSUE THREAD

**Issue by 1366613** [rs/zerolog#622]
I'm not sure what's wrong with this code.

```go
package main

import (
	"fmt"
	"github.com/rs/zerolog"
	"github.com/rs/zerolog/diode"
	"os"
	"time"
)

func main() {
	wr := diode.NewWriter(os.Stdout, 1000, 10*time.Millisecond, func(missed int) {
		fmt.Printf("Logger Dropped %d messages", missed)
	})
	log := zerolog.New(wr).With().Caller().Timestamp().Logger()
	log.Fatal().Msg("test")
	time.Sleep(10 * time.Second)
}

```

**Comment by dtuck9** [comment]
I am also hitting this w/ `github.com/rs/zerolog v1.31.0`:

```
	zerolog.SetGlobalLevel(zerolog.InfoLevel)
	if debug {
		zerolog.SetGlobalLevel(zerolog.DebugLevel)
	}
	logger := zerolog.New(os.Stdout).Hook(logging.Hook{})
....
logger.Fatal().Msgf("Invalid environment [%s] selected. Select from [%v]", env, validEnvs)
```

And I encountered it with and without the `Hook`. 

All other log lines are working as expected.

**Comment by gmbuell** [comment]
Fixed in #634
